File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/marker/BarChartMarkerView.java
package io.github.zwieback.familyfinance.business.chart.marker;

import android.annotation.SuppressLint;
import android.content.Context;
import android.widget.TextView;
import androidx.annotation.NonNull;

import com.github.mikephil.charting.components.MarkerView;
import com.github.mikephil.charting.data.Entry;
import com.github.mikephil.charting.formatter.IAxisValueFormatter;
import com.github.mikephil.charting.highlight.Highlight;
import com.github.mikephil.charting.utils.MPPointF;

import io.github.zwieback.familyfinance.R;

@SuppressLint("ViewConstructor")
public class BarChartMarkerView extends MarkerView {

    @NonNull
    private final TextView markerContent;
    @NonNull
    private final IAxisValueFormatter xAxisValueFormatter;
    @NonNull
    private final IAxisValueFormatter yAxisValueFormatter;

    public BarChartMarkerView(@NonNull Context context,
                              @NonNull IAxisValueFormatter xAxisValueFormatter,
                              @NonNull IAxisValueFormatter yAxisValueFormatter) {
        super(context, R.layout.chart_bar_marker_view);
        this.xAxisValueFormatter = xAxisValueFormatter;
        this.yAxisValueFormatter = yAxisValueFormatter;
        this.markerContent = findViewById(R.id.marker_content);
    }

    @SuppressLint("SetTextI18n")
    @Override
    public void refreshContent(Entry e, Highlight highlight) {
        markerContent.setText("x: " + xAxisValueFormatter.getFormattedValue(e.getX(), null) +
                "; y: " + yAxisValueFormatter.getFormattedValue(e.getY(), null));
        super.refreshContent(e, highlight);
    }

    @Override
    public MPPointF getOffset() {
        return new MPPointF(-(getWidth() / 2.0f), -getHeight());
    }
}

File: app/src/main/java/io/github/zwieback/familyfinance/business/chart/marker/BarChartMarkerView.kt
package io.github.zwieback.familyfinance.business.chart.marker

import android.annotation.SuppressLint
import android.content.Context
import android.widget.TextView
import com.github.mikephil.charting.components.MarkerView
import com.github.mikephil.charting.data.Entry
import com.github.mikephil.charting.formatter.ValueFormatter
import com.github.mikephil.charting.highlight.Highlight
import com.github.mikephil.charting.utils.MPPointF
import io.github.zwieback.familyfinance.R

@SuppressLint("ViewConstructor")
class BarChartMarkerView(
    context: Context,
    private val xAxisValueFormatter: ValueFormatter,
    private val yAxisValueFormatter: ValueFormatter
) : MarkerView(context, R.layout.chart_bar_marker_view) {

    private val markerContent: TextView = findViewById(R.id.marker_content)

    @SuppressLint("SetTextI18n")
    override fun refreshContent(e: Entry, highlight: Highlight?) {
        markerContent.text = "x: ${xAxisValueFormatter.getFormattedValue(e.x)}" +
                "; y: ${yAxisValueFormatter.getFormattedValue(e.y)}"
        super.refreshContent(e, highlight)
    }

    override fun getOffset(): MPPointF {
        return MPPointF(-(width / 2.0f), (-height).toFloat())
    }
}
